editor/src/clj/editor/workspace.clj

```diff
@@ -449,6 +452,25 @@ ordinary paths."
        (find-resource workspace path evaluation-context)
        (file-resource workspace path evaluation-context)))))
 
+(defn make-proj-path->resource-fn [workspace evaluation-context]
+  ;; Both :root and :editable-proj-path? are properties.
+  (let [project-directory-path (g/valid-node-value workspace :root evaluation-context)
+        editable-proj-path? (g/valid-node-value workspace :editable-proj-path? evaluation-context)
+        resources-by-proj-path (g/tx-cached-node-value workspace :resource-map evaluation-context)
+
+        make-missing-file-resource
+        (fn/memoize
+          (fn make-missing-file-resource [proj-path]
+            (let [file (io/file (str project-directory-path proj-path))]
+              (resource/make-file-resource workspace project-directory-path file [] editable-proj-path?))))]
+
+    (assert (not (g/error? resources-by-proj-path)))
+    (assert (map? resources-by-proj-path))
+    (fn proj-path->resource [proj-path]
+      (when-not (coll/empty? proj-path)
+        (or (resources-by-proj-path proj-path)
+            (make-missing-file-resource proj-path))))))
+
```

Comment:
Unrelated fix. I discovered we had not fully transitioned to in-transaction `proj-path` resolution introduced in #10208 here.